dev/release/README.md

```diff
@@ -223,7 +223,7 @@ Here is my vote:
 +1
 
 [1]: https://github.com/apache/datafusion/tree/a5dd428f57e62db20a945e8b1895de91405958c4
-[2]: https://dist.apache.org/repos/dist/dev/arrow/apache-arrow-datafusion-5.1.0
+[2]: https://dist.apache.org/repos/dist/dev/arrow/apache-datafusion-5.1.0
```

Comment:
This isn't quite right (this path doesn't exist) but I think it will be fixed when we update the release scripts